drivers/gles3/rasterizer_canvas_gles3.h

```diff
@@ -36,11 +36,127 @@
 
 class RasterizerCanvasGLES3 : public RasterizerCanvasBaseGLES3, public RasterizerCanvasBatcher<RasterizerCanvasGLES3, RasterizerStorageGLES3> {
 	friend class RasterizerCanvasBatcher<RasterizerCanvasGLES3, RasterizerStorageGLES3>;
+	friend class Buffers;
 
 private:
-	struct BatchGLData {
-		// for batching
-		GLuint batch_vertex_array[5];
+	struct VAOArray {
+		GLuint array[5];
+	};
+
+	// for batching
+	class BatchGLData {
+	public:
+		BatchGLData(RasterizerCanvasGLES3 &rasterizer) :
+				rasterizer(rasterizer) {}
+
+		inline size_t size() const {
+			return current_size;
+		}
+
+		inline void request(size_t new_size) {
+			// #1 no hysteresis
+			//resize(new_size);
+
+			// #2 no hystersis, no decrement
+			//if (new_size > current_size) {
+			//	resize(new_size);
+			//}
+
+			// #3 hysteresis, no decrement
+			//const float t = 0.5;
+			//hysteresis = hysteresis * (1.0f - t) + new_size * t;
+			//new_size = (size_t)hysteresis;
+			//if (new_size > current_size) {
+			//	resize(new_size);
+			//}
+
+			// #4 hysteresis, decrement
+			//const float t = 0.01;
+			//hysteresis = hysteresis * (1.0f - t) + new_size * t;
+			//new_size = (size_t)hysteresis;
+			//resize(new_size);
+
+			// #5 hysteresis, decrement, instantincrement
+			if (new_size < current_size) {
+				const float t = 0.01;
+				hysteresis = hysteresis * (1.0f - t) + new_size * t;
+				new_size = (size_t)hysteresis;
+			} else {
+				hysteresis = (float)new_size;
+			}
+			resize(new_size);
+		}
```

Comment:
Why is there hysteresis in GLES3 but not in GLES2?

Reply:
We will add hysteresis for GLES2 as well.